### DISCUSSION TITLE
How to change asset type for inline nested loaders?

### DISCUSSION THREAD

**Discussion by loc** [webpack/webpack#15101] [Q&A]
I'm having the exact same issue as @rb-mwindh was having in [this issue](https://github.com/webpack/webpack/issues/12531).

To recap, Angular generates a highly opinionated webpack config that they make difficult to change, so when doing anything that Angular doesn't expect you to be doing, there are problems. I'm trying to apply some css via style-loader and css-loader, but Angular sets the asset type of all .css imports as `asset/resource`. [The solution](https://github.com/webpack/webpack/issues/12531#issuecomment-947760871) posted in the issue above uses an internal `!=!` syntax like this to change the resource type so that the loader output is evaluated: 

```
import styles from 'style.css.webpack[javascript/auto]!=!!!style-loader?injectType=lazyStyleTag!css-loader!some-npm-module/style.css`;
```

This works great for the style-loader, but the output of the nested css-loader is still being treated as an `asset/resource`. Is it possible to apply the same trick to change the type for css-loader too? My attempts at guessing at the syntax have so far failed. Thanks!

**Comment by rb-mwindh** [comment]
Hi Ioc,
I was able to find a solution to my issue as described [here](https://stackoverflow.com/questions/69614700/import-css-file-into-es6-returns-string-instead-of-object).

On the accepted answer you'll find my analysis to the whole issue and 2 solution concepts.

Although I would have preferred the inline approach (1), I didn't want to use an undocumented and hard-to-read syntax which is likely to change in the future.

Thus, I resembled to setting up a customized webpack config (2).
Please note, that I'm not setting it up from scratch. Instead I provide a function that 

- searches for the correct ruleset
- excludes a certain filename pattern
- adds a custom rule for that filename pattern

This solution works like a charm. The downside of this approach is that I need to stick to a naming convention.

In my angular apps I used [@angular-builders/custom-webpack](https://www.npmjs.com/package/@angular-builders/custom-webpack) to set up my webpack configurator.

Please let me know if you need further instructions.

**Reply by loc** [reply]
Yeah, fair. I was trying to avoid `@angular-builders/custom-webpack` because in my testing it's slower than the stock builder, but I guess I just have to admit defeat and move on.

Thanks for the reply and your original work in figuring this out — it was driving me up a wall!
